### PR TITLE
Add ftjam/2.5.2 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **jam/2.5.2**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/jam/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    settings = "os", "arch", "compiler", "build_type"
+
+    def build(self):
+        for f in ("header.h", "main.c", "source.c", "Jamfile"):
+            shutil.copy(os.path.join(self.source_folder, f),
+                        os.path.join(self.build_folder, f))
+        if not tools.cross_building(self.settings):
+            assert os.path.isfile(os.environ["JAM"])
+
+            self.run(os.environ["JAM"])
```
Comment (madebr) [inline recipes/jam/all/test_package/conanfile.py:16]:
```suggestion
            self.run(os.environ["JAM"], run_environment=True)
```

**Comment by SSE4** [comment]
what is the relation between this proposed `Jam` and tools like `boost build`, `boost jam` and `boost b2`?
/cc @grafikrobot

**Comment by madebr** [comment]
@SSE4 
See [boost's documentation](https://www.boost.org/doc/libs/1_31_0/tools/build/jam_src/index.html#introduction): boost's bjam is based on this one.
Maybe this one should be renamed to `ftjam`.
Boost's jam should be named `bjam`.

**Comment by SSE4** [comment]
yes, I think `b2` is a second version of `boost jam` (`bjam`). @grafikrobot please correct me I am wrong
for `ftjam` or just `jam`, do you have some projects which require it to be built? can they be built with `b2` instead of `ftjam`?
as documentation claims that `Boost.Jam (BJam)  is a build tool based on FTJam, which in turn is based on Perforce Jam. It contains significant improvements made to facilitate its use in the Boost Build System, but should be backward compatible with Perforce Jam.`

**Comment by SSE4** [comment]
okay, thanks for the clarification, then it makes total sense to have both `b2` and `ftjam` here. please rename recipe to `ftjam`.

File: recipes/jam/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from contextlib import contextmanager
+import os
+
+
+class JamConan(ConanFile):
+    name = "jam"
```
Comment (SSE4) [inline recipes/jam/all/conanfile.py:7]:
```suggestion
    name = "ftjam"
```

File: recipes/jam/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    description = "Jam is a small open-source build tool that can be used as a replacement for Make."
+    topics = ("conan", "jam", "build", "make")
```
Comment (SSE4) [inline recipes/jam/all/conanfile.py:9]:
```suggestion
    topics = ("conan", "jam", "ftjam", "build", "make")
```

File: recipes/jam/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from contextlib import contextmanager
+import os
+
+
+class JamConan(ConanFile):
```
Comment (SSE4) [inline recipes/jam/all/conanfile.py:6]:
```suggestion
class FtJamConan(ConanFile):
```

**Comment by madebr** [comment]
All os+compiler+build_type combinations.
I'll have to cull some.

@jgsogo
Another one for the collection 😄

File: recipes/ftjam/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        self._autotools.libs = []
+        # The configure MUST be run inside this directory
+        with tools.chdir(os.path.join(self.build_folder, self._source_subfolder, "builds", "unix")):
+            self._autotools.configure()
+        return self._autotools
+
+    def build(self):
+        if self.settings.build_type != "Release":
```
Comment (uilianries) [inline recipes/ftjam/all/conanfile.py:58]: Is there any reason for using this under build() and not configure() ?

Comment (madebr) [reply recipes/ftjam/all/conanfile.py:58]: I did it for the same reason as in the recent ninja update by eric

Comment (uilianries) [reply recipes/ftjam/all/conanfile.py:58]: Agreed!

**Pull request merged**